test/integration/puppetlabs/services/jruby/class_info_test.clj

```diff
@@ -0,0 +1,242 @@
+                "type" "Integer"}
+               {"name" (str class "_c"),
+                "type" "String",
+                "default_literal" "c default value"}]})
+
+(defn expected-manifests-info
+  [manifests]
+  (into {}
+        (apply concat
+               (for [[dir names] manifests]
+                 (do
+                   (for [name names]
+                     [(.getAbsolutePath
+                        (fs/file dir
+                                 "manifests"
+                                 (str name ".pp")))
+                      [(expected-class-info name)
+                       (expected-class-info
+                         (str name "2"))]]))))))
+
+(deftest ^:integration class-info-test
+  (testing "class info properly enumerated for"
+    (let [pool (LinkedBlockingDeque. 1)
+          code-dir (ks/temp-dir)
+          conf-dir (ks/temp-dir)
+          config (jruby-testutils/jruby-puppet-config
+                   {:master-code-dir (.getAbsolutePath code-dir)
+                    :master-conf-dir (.getAbsolutePath conf-dir)})
+          instance (jruby-internal/create-pool-instance!
+                     pool 0 config #() nil)
+          jruby-puppet (:jruby-puppet instance)
+          container (:scripting-container instance)
+
+          _ (create-file (fs/file conf-dir "puppet.conf")
+                         "[main]\nenvironment_timeout=unlimited\nbasemodulepath=$codedir/modules\n")
+
+          env-dir (fn [env-name]
+                    (fs/file code-dir "environments" env-name))
+          env-1-dir (env-dir "env1")
+          env-1-dir-and-manifests [env-1-dir ["foo" "bar"]]
+          _ (create-env env-1-dir-and-manifests)
+
+          env-2-dir (env-dir "env2")
+          env-2-dir-and-manifests [env-2-dir ["baz" "bim" "boom"]]
+          _ (create-env env-2-dir-and-manifests)
+
+          env-1-mod-dir (fs/file env-1-dir "modules")
+          env-1-mod-1-dir-and-manifests [(fs/file env-1-mod-dir
+                                                  "envmod1")
+                                         ["envmod1baz" "envmod1bim"]]
+          _ (gen-classes env-1-mod-1-dir-and-manifests)
+          env-1-mod-2-dir (fs/file env-1-mod-dir "envmod2")
+          env-1-mod-2-dir-and-manifests [env-1-mod-2-dir
+                                         ["envmod2baz" "envmod2bim"]]
+          _ (gen-classes env-1-mod-2-dir-and-manifests)
+
+          env-3-dir-and-manifests [(env-dir "env3") ["dip" "dap" "dup"]]
+
+          base-mod-dir (fs/file code-dir "modules")
+          base-mod-1-and-manifests [(fs/file base-mod-dir "basemod1")
+                                    ["basemod1bap"]]
+          _ (gen-classes base-mod-1-and-manifests)
+
+          bogus-env-dir (env-dir "bogus-env")
+          _ (create-env [bogus-env-dir []])
+          _ (gen-classes [bogus-env-dir ["envbogus"]])
+          _ (gen-classes [(fs/file base-mod-dir "base-bogus") ["base-bogus1"]])
+
+          get-class-info (fn []
```

Comment:
I left these two functions in the let block rather than separating them because they close over `jruby-puppet` and didn't want to have to instead repeat `jruby-puppet` as an argument down where they are used inside of the testing sub-blocks.